### PR TITLE
REFACTOR: aedt_file_management file

### PR DESCRIPTION
**Pull request by Marwane-20**
## Description

This pull request refactors the aedt_file_management file and adds comprehensive tests.
@SMoraisAnsys for review.

---

#### `create_output_folder`
- Simplified directory creation using `os.makedirs(..., exist_ok=True)`.
- Improved variable naming for clarity.

#### `change_objects_visibility`
- Switched file handling from binary to text mode with explicit UTF‑8 encoding.
- Added robust error handling with try/except blocks.
- Aligned return values with documented behavior (returns True on success, False on error).

#### `change_model_orientation`
- Applied similar improvements in file handling and error management.
- Uses `orientation.get(bottom_dir, "")` for safer dictionary lookup.

#### Tests Addition:
- Added unit, integration, and system tests for all functions in aedt_file_management.

---

**IMPORTANT Note:**
Assumes files are UTF‑8 encoded and that silently handling an invalid `bottom_dir` is acceptable.

## Issue linked
Issue: #5756

## Checklist
- [X] Tested changes locally.
- [X] Followed coding style guidelines.
- [X] Added appropriate tests.
- [X] Reviewed changes before submitting.
- [X] Linked the issue.
- [X] Agreed with the Contributor License Agreement.

### CODE REVIEW

**Comment by Marwane-20** [comment]
@ansys-reviewer-bot review

File: src/ansys/aedt/core/application/aedt_file_management.py [hunk 1]
```diff
@@ -95,22 +95,20 @@ def create_output_folder(ProjectDir):
...
-        os.mkdir(OutputPath)
-    if not os.path.exists(PicturePath):
-        os.mkdir(PicturePath)
-    if not os.path.exists(ResultsPath):
-        os.mkdir(ResultsPath)
-    return PicturePath, ResultsPath
+    os.makedirs(output_path, exist_ok=True)
+    os.makedirs(picture_path, exist_ok=True)
+    os.makedirs(results_path, exist_ok=True)
+    return picture_path, results_path
```
[the remark this answers is not in the document]
Comment (SMoraisAnsys) [reply src/ansys/aedt/core/application/aedt_file_management.py:111]: There is a typo in the proposed changes but I agree.

File: src/ansys/aedt/core/application/aedt_file_management.py [hunk 2]
```diff
@@ -133,29 +131,36 @@ def change_objects_visibility(origfile, solid_list):
...
+                n.write(s)
+            # Renaming files and deleting temporary file
+            os.remove(origfile)
+            os.rename(newfile, origfile)
+            return True
+        except Exception as e:
+            # Cleanup temporary file if exists.
+            if os.path.exists(newfile):
+                os.remove(newfile)
+            print("change_objects_visibility: Error encountered -", e)
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/application/aedt_file_management.py:159]: Please use logging here instead of printing.

File: src/ansys/aedt/core/application/aedt_file_management.py [hunk 3]
```diff
@@ -133,29 +131,36 @@ def change_objects_visibility(origfile, solid_list):
...
-            n.write(str(s))
-
-        # renaming files and deleting temp
-
-        os.remove(origfile)
-        os.rename(newfile, origfile)
-
+        try:
+            # Using text mode with explicit encoding instead of binary mode.
+            with open(origfile, "r", encoding="utf-8") as f, open(newfile, "w", encoding="utf-8") as n:
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/application/aedt_file_management.py:136]: Same comment about "utf-8"

File: src/ansys/aedt/core/application/aedt_file_management.py [hunk 4]
```diff
@@ -200,25 +205,32 @@ def change_model_orientation(origfile, bottom_dir):
...
-            # Writing file content
-            n.write(str(s))
-
-        # Renaming files and deleting temp
-        os.remove(origfile)
-        os.rename(newfile, origfile)
-
+        try:
+            # Using text mode with explicit encoding.
+            with open(origfile, "r", encoding="utf-8") as f, open(newfile, "w", encoding="utf-8") as n:
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/application/aedt_file_management.py:210]: Same comment about "utf-8"

File: tests/integration/test_aedt_file_management.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    expected_results = os.path.join(str(output_dir), "Results")
+    assert picture_path == expected_picture
+    assert results_path == expected_results
+    assert os.path.isdir(picture_path)
+    assert os.path.isdir(results_path)
+
+
+def test_integration_change_objects_visibility(tmp_path):
+    # Create a temporary AEDT file with multiline content.
+    content = "$begin 'EditorWindow'\n" "Header info\n" "Drawings[old_value]\n" "Footer info\n" "   $end 'EditorWindow'"
```
Comment (SMoraisAnsys) [inline tests/integration/test_aedt_file_management.py:80]: Can you homogenize the way of creating content ? See next test

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
Thanks for the contribution. I've added some minor comments and the review bot also did, can you address them ?

Also, could you tell me what's the value of adding all the tests you did ? I've got the feeling that they are mainly "unit" tests and there are multiple duplicates of the same things being tested in different folders but in the same way.

File: src/ansys/aedt/core/application/aedt_file_management.py [hunk 6]
```diff
@@ -49,12 +49,12 @@ def read_info_fromcsv(projdir, name):
     """
 
     filename = projdir + "//" + name
-    listmcad = []
-    with open_file(filename, "rb") as csvfile:
+    rows = []
+    with open_file(filename, "r", encoding="utf-8") as csvfile:
```
[the remark this answers is not in the document]
Comment (MaxJPRey) [reply src/ansys/aedt/core/application/aedt_file_management.py:53]: Good catch @SMoraisAnsys regarding the encoding.

**Comment by MaxJPRey** [comment]
Thanks @Marwane-20 for the contribution.

**Comment by Marwane-20** [comment]
@SMoraisAnsys I see, just based on the parent issue #5504 there was mention of the three layers of test:
>Improve Code Coverage: Increase the percentage of code covered by automated tests (unit, integration, and system tests). Reach 90%.
So i thought it was mandatory.

**Comment by Samuelopez-ansys** [comment]
I cherry-picked your commits here: #5879

**Comment by SMoraisAnsys** [comment]
@Marwane-20 A refactoring was performed on the repo and part of the files you were working on got involved. @Samuelopez-ansys cherry-pick your changes, we'll review them and handle the following. Thanks for the contribution

**Comment by Marwane-20** [comment]
@SMoraisAnsys, @Samuelopez-ansys sounds good, thanks !
